nuclear_simulator/simulator/state/state_manager.py

```
@@ -0,0 +1,1910 @@
+        """
+        import time
+        start_time = time.time()
+        
+        # Start with datetime
+        row_data = {'time': current_datetime}
+        
+        # Collect from all providers
+        for provider, category in self.providers:
+            try:
+                # Try StateProvider protocol first (get_current_state)
+                if hasattr(provider, 'get_current_state') and callable(getattr(provider, 'get_current_state')):
```

Ruff (0.11.9)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)